### PR TITLE
Some more ZF

### PR DESCRIPTION
**Pull request by robertbastian**

### CODE REVIEW

**Review by sffc — COMMENTED** [review]
LGTM but please wait for @Manishearth's review (and fix CI)

File: provider/macros/src/lib.rs [hunk 1]
```diff
@@ -21,7 +21,7 @@ mod tests;
 /// for use in a `DataStruct`. It does the following things:
 ///
 /// - `Apply #[derive(Yokeable, ZeroFrom)]`. The `ZeroFrom` derive can
-///    be customized with `#[zerofrom(cloning_zf)]` as needed
+///    be customized with `#[zerofrom(clone)]` on non-ZeroFrom fields.
 ///
 /// In addition, the attribute can be used to implement `DataMarker` and/or `ResourceMarker`
 /// by adding symbols with optional key strings:
```
[the remark this answers is not in the document]
Comment (robertbastian) [reply provider/macros/src/lib.rs:27]: From the second commit, but this is WIP

**Review by Manishearth — DISMISSED** [review]
lgtm too, good call on `zerofrom(clone)`

File: utils/yoke/src/yokeable.rs [hunk 2]
```diff
@@ -296,3 +296,30 @@ unsafe impl<'a, T: 'static + ?Sized> Yokeable<'a> for &'static T {
...
+        self
+    }
+    #[inline]
+    fn transform_owned(self) -> alloc::vec::Vec<T> {
+        // Doesn't need unsafe: `'a` is covariant so this lifetime cast is always safe
+        self
+    }
+    #[inline]
+    unsafe fn make(from: alloc::vec::Vec<T>) -> Self {
+        mem::transmute(from)
```
Comment (Manishearth) [inline utils/yoke/src/yokeable.rs:315]: nit: can just be `from`

File: utils/yoke/src/yokeable.rs [hunk 3]
```diff
@@ -296,3 +296,30 @@ unsafe impl<'a, T: 'static + ?Sized> Yokeable<'a> for &'static T {
...
+    }
+    #[inline]
+    fn transform_mut<F>(&'a mut self, f: F)
+    where
+        F: 'static + for<'b> FnOnce(&'b mut Self::Output),
+    {
+        // Cast away the lifetime of Self
+        unsafe { f(mem::transmute::<&'a mut Self, &'a mut Self::Output>(self)) }
```
Comment (Manishearth) [inline utils/yoke/src/yokeable.rs:323]: nit: no need to transmute i think

**Review by Manishearth — APPROVED** [review]

**Review by Manishearth — APPROVED** [review]

**Pull request merged**